source/darwin/darwin_pki_utils.c

```diff
@@ -517,39 +579,214 @@ int aws_secitem_import_cert_and_key(
     key_attributes =
         CFDictionaryCreateMutable(cf_alloc, 0, &kCFTypeDictionaryKeyCallBacks, &kCFTypeDictionaryValueCallBacks);
     CFDictionaryAddValue(key_attributes, kSecAttrKeyClass, kSecAttrKeyClassPrivate);
+    // TODO Do we need this?
     CFDictionaryAddValue(key_attributes, kSecAttrKeyType, key_type);
+
 #ifdef AWS_SECITEM_FILEBASED_KEYCHAIN
     /* Target file-based keychain instead of data protection keychain. */
     CFDictionaryAddValue(key_attributes, kSecUseDataProtectionKeychain, kCFBooleanFalse);
 #endif
 
+    /*
+     * Try to parse a user-provided private key into a SecKeyRef. On this step, the private key won't be added
+     * to keychain yet.
+     */
     key_ref = SecKeyCreateWithData(key_data, key_attributes, &error);
     if (error) {
         char description_buffer[256];
         aws_get_core_foundation_error_description(error, description_buffer, sizeof(description_buffer));
-        AWS_LOGF_ERROR(AWS_LS_IO_PKI, "Failed creating SecKeyRef from key_data: %s", description_buffer);
+        AWS_LOGF_WARN(AWS_LS_IO_PKI, "Failed importing private key using SecItem: %s", description_buffer);
+
+        /*
+         * If parsing with SecItem fails, we fall back to trying to add the private key via SecKeychain API.
```

Comment:
Either we should rename `s_import_ecc_key_into_keychain` to not be "ecc" specific or we should call out that we are attempting to import an ecc private key.